### ISSUE TITLE
Sample IRobotsTxtProvider Doesn't Work

### ISSUE THREAD

**Issue by RichMercer** [karl-sjogren/robots-txt-middleware#6]
The sample IRobotsTxtProvider CoolRobotsTxtProvider renders the type of class in the resulting robots.txt rather than the configured values.

`builder.AllowAll().Build()` returns `RobotsTxtOptions` which doesn't have an override for `ToString()` so it just returns the type as a string. Really a call to `builder.AllowAll().Build().Build()` is required to generate the correct output, but the `Build()` method on `RobotsTxtOptions` is internal so can't be accessed. I can't see a way to implement a custom provider and have it build the correct output.

**Comment by KenBrannigan** [comment]
I am having the same issue. When will this be fixed?

**Comment by karl-sjogren** [comment]
Oh sorry, I must have accidently turned of notifications for this repo. I'll look into this in the coming days.

**Comment by karl-sjogren** [comment]
I've released version 2.1.1 of the library and updated the code for the sample as well. The `RobotsTxtOptions` class now overrides `ToString()` to get the generated file content (this was previously an internal method).

The sample now looks like this.

```csharp
public class CoolRobotsTxtProvider : IRobotsTxtProvider {
    private readonly CoolContext _context;

    public CoolRobotsTxtProvider(CoolContext context) {
        _context = context;
    }

    public async Task<RobotsTxtResult> GetResultAsync(CancellationToken cancellationToken) {
        var settings = await _context.Settings.FirstAsync();

        var builder = new RobotsTxtOptionsBuilder();

        RobotsTxtOptions options;
        if(settings.AllowAllRobots)
            options = builder.AllowAll().Build();
        else
            options = builder.DenyAll().Build();

        var content = options.ToString();
        var buffer = Encoding.UTF8.GetBytes(content).AsMemory();
        return new RobotsTxtResult(buffer, settings.RobotsTxtMaxAge);
    }
}
```

I've also made is so that the example is built along with the application through the `RobotsTxt.Samples` project so I should notice if I break it again (I believe I wrote the sample for one of the 2.0 preview releases or something and then forgot about it since it lived in the readme).
